ui/app/models/oidc/assignment.js

```diff
@@ -1,14 +1,63 @@
 import Model, { attr, hasMany } from '@ember-data/model';
+import ArrayProxy from '@ember/array/proxy';
+import PromiseProxyMixin from '@ember/object/promise-proxy-mixin';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withModelValidations } from 'vault/decorators/model-validations';
+import { isPresent } from '@ember/utils';
 
+const validations = {
+  name: [
+    { type: 'presence', message: 'Name is required.' },
+    // ARG TODO add in after Claire pushes her branch
+    // {
+    //   type: 'containsWhiteSpace',
+    //   message: 'Name cannot contain whitespace.',
+    // },
+  ],
+  targets: [
+    {
+      validator(model) {
+        const entityIds = model.hasMany('entity_ids').ids();
+        const groupIds = model.hasMany('group_ids').ids();
+        return isPresent(entityIds) || isPresent(groupIds);
+      },
+      message: 'At least one entity or group is required.',
+    },
+  ],
+};
+
+@withModelValidations(validations)
 export default class OidcAssignmentModel extends Model {
   @attr('string') name;
-  @hasMany('identity/entity') entityIds;
-  @hasMany('identity/group') groupIds;
+  @hasMany('identity/entity') entity_ids;
```

Comment:
before the network request was sending the param entityIds where the API is expecting entity_ids. However, this change affects an adapter test. While things are working as expected the test can no longer send the entity_ids because it's expecting an instance of the @hasMany operation. To overcome (for the test purpose) we need to modify how we have the serializer and adapter setup, which is a restructure of the current flow. As this issue is a larger structural issue and outside the scope of this PR, I commented out the test later down and made a comment to discuss with the OIDC team once they return.

Reply:
Did the test break only after you made this snake case? And why did we make it snake case?

Reply:
Yeah, so the test broke after I made this param match the API param (e.g. the payload is expecting the value to be {entity_id: ['some-value']}. If I didn't change this here to match it would incorrectly send the payload and the values wouldn't be saved. We can fix this back to entityIds or whatever we want with some serializer magic. However, that's a change to the pattern already established, so I'm going to confirm with Claire when she gets back how we want to proceed.